Sources/Logging/Destinations/Log+ConsoleLogDestination.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  Log+ConsoleLogDestination.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 07/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public extension Log {
+
+    public enum ConsoleLogDestinationError: Swift.Error {
+        case itemFormatFailed(Log.Item, Swift.Error)
+    }
+
+    public class ConsoleLogDestination<ItemFormatter: LogItemFormatter>: LogDestination
+    where ItemFormatter.Output == String {
+
+        public typealias OutputClosure = ((Level, String) -> Void)
+        public typealias LogMetadataClosure = ([AnyHashable : Any]) -> (Level, String)
+
+        public let formatter: ItemFormatter
+
+        public let minLevel: Level
+
+        private let output: OutputClosure
+        private let logMetadata: LogMetadataClosure?
+
+        // MARK: - Lifecycle
+
+        public init(formatter: ItemFormatter,
+                    minLevel: Level = .error,
+                    output: @escaping OutputClosure = { (level, message) in print(message) },
+                    logMetadata: LogMetadataClosure? = nil) {
+
+            self.minLevel = minLevel
+            self.formatter = formatter
+            self.output = output
+            self.logMetadata = logMetadata
+        }
+
+        // MARK: - Public methods
+
+        public func write(item: Log.Item, failure: @escaping (Error) -> Void) {
```

Comment:
What do you think about specialize the error type as well? Like, `class ConsoleLogDestination<ItemFormatter: LogItemFormatter, Error: ConsoleLogDestinationError>`? Or would it cause a pain in the 🍑?

Reply:
It's doable, but I don't see any clear benefit in doing so. 🤔 Can you please elaborate?

The error handling on the `Logger` has to be generic (i.e. work on `Error` alone), and each destination should have (and already has) its own error type according to its implementation  (if needed).

Alternatively, we could create a generic `LogDestinationError` that all destinations would use, with errors for the main operations, like:

```swift
enum LogDestinationError: Error {
    case itemFormat(Log.Item, Error?)
    case write(Log.Item, Error?)
    case setMetadata([AnyHashable : Any], Error?)
    case removeMetadata([AnyHashable], Error?)
    case other(Error?)
}
```

But once again, I don't see any practical value in this because (for now) we're using a `throws` based API and as such on the `Logger` we wouldn't have type safe errors 😓

Reply:
The advantages do not justify the disadvantages of this, especially because this failure closure has a really little impact on the logging system. Nevermind. 🙇‍♂️